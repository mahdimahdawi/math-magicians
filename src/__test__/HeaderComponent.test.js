import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import Header from '../components/header';

describe('Testing Header Component', () => {
  test('Testing that all the links are present and rendered', () => {
    render(
      <Router>
        <Header />
      </Router>
    );

    expect(screen.getByText('Home')).toMatchSnapshot();
    expect(screen.getByText('Calculator')).toMatchSnapshot();
    expect(screen.getByText('Quote')).toMatchSnapshot();
  });

  test('navigate to Home Page', () => {
    render(
      <Router>
        <Header />
      </Router>
    );

    fireEvent.click(screen.getByText('Home'));

    expect(screen.getByText('Home')).toMatchInlineSnapshot(`
<a
  aria-current="page"
  class="menu-link active"
  href="/"
>
  Home
</a>
`);
  });

  test('navigate to Calculator Page', () => {
    render(
      <Router>
        <Header />
      </Router>
    );

    fireEvent.click(screen.getByText('Calculator'));

    expect(screen.getByText('Calculator')).toMatchInlineSnapshot(`
<a
  aria-current="page"
  class="menu-link active"
  href="/calculator"
>
  Calculator
</a>
`);
  });

  test('navigate to Quote Page', () => {
    render(
      <Router>
        <Header />
      </Router>
    );

    fireEvent.click(screen.getByText('Quote'));

    expect(screen.getByText('Quote')).toMatchInlineSnapshot(`
<a
  aria-current="page"
  class="menu-link active"
  href="/quote"
>
  Quote
</a>
`);
  });
});