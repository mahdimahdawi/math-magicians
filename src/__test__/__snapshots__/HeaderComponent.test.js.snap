// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header Component Testing that all the links are present and rendered 1`] = `
<a
  aria-current="page"
  class="menu-link active"
  href="/"
>
  Home
</a>
`;

exports[`Testing Header Component Testing that all the links are present and rendered 2`] = `
<a
  class="menu-link"
  href="/calculator"
>
  Calculator
</a>
`;

exports[`Testing Header Component Testing that all the links are present and rendered 3`] = `
<a
  class="menu-link"
  href="/quote"
>
  Quote
</a>
`;
